spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/source/TestCompressionSettings.java

```diff
@@ -0,0 +1,181 @@
+          optional(1, "id", Types.IntegerType.get()), optional(2, "data", Types.StringType.get()));
+
+  @Rule public TemporaryFolder temp = new TemporaryFolder();
+
+  @Parameterized.Parameters(name = "format = {0}, properties = {1}")
+  public static Object[] parameters() {
+    return new Object[] {
+      new Object[] {"parquet", ImmutableMap.of(COMPRESSION_CODEC, "zstd", COMPRESSION_LEVEL, "1")},
+      new Object[] {"parquet", ImmutableMap.of(COMPRESSION_CODEC, "gzip")},
+      new Object[] {
+        "orc", ImmutableMap.of(COMPRESSION_CODEC, "zstd", COMPRESSION_STRATEGY, "speed")
+      },
+      new Object[] {
+        "orc", ImmutableMap.of(COMPRESSION_CODEC, "zstd", COMPRESSION_STRATEGY, "compression")
+      },
+      new Object[] {"avro", ImmutableMap.of(COMPRESSION_CODEC, "snappy", COMPRESSION_LEVEL, "3")}
+    };
+  }
+
+  @BeforeClass
+  public static void startSpark() {
+    TestCompressionSettings.spark = SparkSession.builder().master("local[2]").getOrCreate();
+  }
+
+  @Parameterized.AfterParam
+  public static void clearSourceCache() {
+    ManualSource.clearTables();
+  }
+
+  @AfterClass
+  public static void stopSpark() {
+    SparkSession currentSpark = TestCompressionSettings.spark;
+    TestCompressionSettings.spark = null;
+    currentSpark.stop();
+  }
+
+  public TestCompressionSettings(String format, ImmutableMap properties) {
+    this.format = FileFormat.fromString(format);
+    this.properties = properties;
+  }
+
+  @Test
+  public void testWriteDataWithDifferentSetting() throws Exception {
+    File parent = temp.newFolder(format.toString());
+    File location = new File(parent, "test");
+    HadoopTables tables = new HadoopTables(CONF);
+    Map<String, String> tableProperties = Maps.newHashMap();
+    tableProperties.put(PARQUET_COMPRESSION, "gzip");
+    tableProperties.put(AVRO_COMPRESSION, "gzip");
+    tableProperties.put(ORC_COMPRESSION, "zlib");
+    Table table =
+        tables.create(SCHEMA, PartitionSpec.unpartitioned(), tableProperties, location.toString());
+    List<SimpleRecord> expectedOrigin = Lists.newArrayList();
+    for (int i = 0; i < 1000; i++) {
+      expectedOrigin.add(new SimpleRecord(1, "hello world" + i));
+    }
+
+    Dataset<Row> df = spark.createDataFrame(expectedOrigin, SimpleRecord.class);
+
+    for (Map.Entry<String, String> entry : properties.entrySet()) {
+      spark.conf().set(entry.getKey(), entry.getValue());
+    }
+
+    df.select("id", "data")
+        .write()
+        .format("iceberg")
+        .option(SparkWriteOptions.WRITE_FORMAT, format.toString())
+        .mode(SaveMode.Append)
+        .save(location.toString());
+
+    List<ManifestFile> manifestFiles = table.currentSnapshot().dataManifests(table.io());
+    try (ManifestReader<DataFile> reader = ManifestFiles.read(manifestFiles.get(0), table.io())) {
+      DataFile file = reader.iterator().next();
+      InputFile inputFile = table.io().newInputFile(file.path().toString());
+      Assert.assertEquals(
```

Comment:
Please follow the contribution guide using `AssertJ` instead. You can refer here: https://iceberg.apache.org/contribute/#assertj

Reply:
Fixed.